akka-remote/src/test/scala/akka/remote/artery/TestContext.scala

```diff
@@ -6,18 +6,29 @@ package akka.remote.artery
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.CopyOnWriteArrayList
 import java.util.concurrent.ThreadLocalRandom
-
 import scala.concurrent.Future
 import scala.concurrent.Promise
 import scala.util.Success
-
 import akka.Done
 import akka.actor.ActorRef
 import akka.actor.Address
 import akka.remote.RemoteActorRef
 import akka.remote.UniqueAddress
 import akka.remote.artery.InboundControlJunction.ControlMessageObserver
 import akka.remote.artery.InboundControlJunction.ControlMessageSubject
+import akka.util.OptionVal
+import akka.actor.InternalActorRef
+
+private[akka] final case class TestInboundEnvelope(
+  recipient:        InternalActorRef,
+  recipientAddress: Address,
+  message:          AnyRef,
+  senderOption:     OptionVal[ActorRef],
+  originUid:        Long) extends InboundEnvelope {
+
+  override def withMessage(message: AnyRef): InboundEnvelope =
+    copy(message = message)
+}
```

Comment:
Don't we want to use the actual envelope rather than an immutable test-version of it? Could we have a nice factory instead of a separate implementation?